File: app/src/main/java/me/ykrank/s1next/view/adapter/delegate/ProgressAdapterDelegate.java
package me.ykrank.s1next.view.adapter.delegate;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import java.util.List;

import me.ykrank.s1next.R;
import me.ykrank.s1next.databinding.ItemProgressBinding;
import me.ykrank.s1next.view.adapter.item.ProgressItem;

public final class ProgressAdapterDelegate extends BaseAdapterDelegate<ProgressItem, ProgressAdapterDelegate.ProgressViewHolder> {

    public ProgressAdapterDelegate(Context context) {
        super(context);
    }

    @NonNull
    @Override
    protected Class<ProgressItem> getTClass() {
        return ProgressItem.class;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ProgressViewHolder(DataBindingUtil.inflate(mLayoutInflater, R.layout.item_progress, parent, false));
    }

    @Override
    public void onBindViewHolderData(ProgressItem item, int position, @NonNull ProgressViewHolder holder, @NonNull List<Object> payloads) {

    }

    static final class ProgressViewHolder extends RecyclerView.ViewHolder {

        private final ItemProgressBinding binding;

        public ProgressViewHolder(ItemProgressBinding binding) {
            super(binding.getRoot());

            this.binding = binding;
        }
    }
}

File: library/src/main/java/com/github/ykrank/androidtools/ui/adapter/delegate/ProgressAdapterDelegate.kt
package com.github.ykrank.androidtools.ui.adapter.delegate

import android.content.Context
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup

import com.github.ykrank.androidtools.R
import com.github.ykrank.androidtools.databinding.ItemProgressBinding
import com.github.ykrank.androidtools.ui.adapter.delegate.item.ProgressItem


class ProgressAdapterDelegate(context: Context) : LibBaseAdapterDelegate<ProgressItem,
        ProgressAdapterDelegate.ProgressViewHolder>(context, ProgressItem::class.java) {

    public override fun onCreateViewHolder(parent: ViewGroup): androidx.recyclerview.widget.RecyclerView.ViewHolder {
        return ProgressViewHolder(DataBindingUtil.inflate(mLayoutInflater, R.layout.item_progress, parent, false))
    }

    override fun onBindViewHolderData(t: ProgressItem, position: Int, holder: ProgressViewHolder, payloads: List<Any>) {

    }

    class ProgressViewHolder(private val binding: ItemProgressBinding) : androidx.recyclerview.widget.RecyclerView.ViewHolder(binding.root)
}
